l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -302,8 +302,15 @@ library Decoder {
 
       // Create the leaf to contain commitments (2 * COMMITMENTS_PER_TX * 020) + nullifiers (2 * NULLIFIERS_PER_TX * 0x20)
       // + new public data writes (8 * 0x40) + contract deployments (2 * 0x60) + logs hashes (2 * 4 * 0x20)
+      ArrayLengths memory constantsLengthsPerBase;
```

Comment:
Think the name should be reconsidered, it is a bit odd. The length seems weird when it is not something that changes but might just be me. Also the `commitmentCount` and such that you get from the `ArrayLengths` just sounds strange to me when like this. 


Also, consider if you actually want to store the count, or if we are better of storing the number of bytes instead. 

If you look at where it is used, it is mostly needing the number of bytes to jump and such so we can skip some later computation if we had those instead. 

If we are storing bytes, we could 
```solidity
// Stack too deep reee
struct MemoryByteConstants {
    uint256 commitmentBytesPerKernel;
    uint256 nullifierBytesPerKernel;
    ...
}
```

Then the baseleaf later is just the sum of those. And we skip a lot of multiplications later in the loop. Can be used for the offsets then as well (around line 440)